keps/sig-instrumentation/1668-log-tracking/README.md

```diff
@@ -0,0 +1,366 @@
+
+The following picture show our design![design](./overview.png)
+
+
+
+we don't have any modifications in kubectl in this design.
+
+### Design of ID propagation in apiserver
+
+**1. Preprocessing handler (othttp handler)**
+
+1.1 Do othttp's original [Extract](https://pkg.go.dev/go.opentelemetry.io/otel/api/propagators#TraceContext.Extract)() to get [SpanContext](https://pkg.go.dev/go.opentelemetry.io/otel/api/trace#SpanContext)
+
+- For request from kubectl, SpanContext is null,  do [Start](https://github.com/open-telemetry/opentelemetry-go/blob/3a9f5fe15f50a35ad8da5c5396a9ed3bbb82360c/sdk/trace/tracer.go#L38)() to start new SpanContext (new trace-id and span-id)
+- For request from controller we can get a valid SpanContext(including trace-id and span-id), do [Start](https://github.com/open-telemetry/opentelemetry-go/blob/3a9f5fe15f50a35ad8da5c5396a9ed3bbb82360c/sdk/trace/tracer.go#L38)() to update the SpanContext (new span-id)
+
+1.2 Chain SpanContext and initial-trace-id to "r.ctx"
+
+- we use r.ctx to propagate those IDs to next handler
+
+**2. inittrace handler**
+
+Implement a new inittrace handler to propagate the initial-trace-id.
+
+2.1 do our new Extract() to get initial-trace-id from request header
+
+- For request from kubectl we can't get initial-trace-id,  chain a empty initial-trace-id to a golang ctx
+- For request from controller we can get initial-trace-id, chain the initial-trace-id to a golang ctx
+
+2.2 chain above golang cxt to r.ctx(updated in above 1.2)
+-  we use r.ctx to propagate those IDs to next handler
+
+**3. Inject SpanContext and initial-trace to request header sent to webhook**
+
+- call othttp's original [Inject](https://pkg.go.dev/go.opentelemetry.io/otel/api/propagators#TraceContext.Inject)() to inject the SpanContext from r.ctx to header
+- call our new Inject() to inject the initial-trace-id from r.ctx to header
+
+### Design of ID propagation (controller)
+
+**1. Extract SpanContext and initial-trace-id from annotation of object to golang ctx**
+
+**2. Propagate golang ctx through objects**
+
+**3. Inject SpanContext and initial-trace to request header sent to apiserver**
+
+- extract SpanContext and initial-trace-id from golang ctx,
+- inject them the header
+
+### Design of Mutating webhook(Out of tree)
+
+**1. Extract SpanContext and initial-trace-id from request's header**
+
+**2. Update SpanContext and initial-trace-id to object**
+
+- if there is initial-trace-id, add trace-id, span-id and initial-trace-id to annotation (This is the case for requests from controller.)
+
+- if there is no initial-trace-id, check the request's operation
+  - if operation is create, copy the trace-id as initial-trace-id, and add trace-id, span-id and initial-trace-id to annotation (This is the case for requests from kubectl create.)
+  - if operation is not create, add trace-id, span-id to annotation (This is the case for requests from kubectl other than create.)
+
+**3. Persist object to etcd**
```

Comment:
I get what you mean, but the mutating webhook doesn't write to etcd.  Maybe just remove this item?

Reply:
Thanks, we will remove this item.